data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/handlers/network/GetListenPort.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.beaconrestapi.handlers.network;
+
+import static tech.pegasys.artemis.beaconrestapi.CacheControlUtils.CACHE_NONE;
+import static tech.pegasys.artemis.beaconrestapi.RestApiConstants.RES_INTERNAL_ERROR;
+import static tech.pegasys.artemis.beaconrestapi.RestApiConstants.RES_OK;
+import static tech.pegasys.artemis.beaconrestapi.RestApiConstants.TAG_NETWORK;
+
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import tech.pegasys.artemis.api.NetworkDataProvider;
+import tech.pegasys.artemis.provider.JsonProvider;
+
+public class GetListenPort implements Handler {
+
+  public static final String ROUTE = "/network/listen_port";
+
+  private final NetworkDataProvider network;
+  private final JsonProvider jsonProvider;
+
+  public GetListenPort(NetworkDataProvider network, JsonProvider jsonProvider) {
+    this.network = network;
+    this.jsonProvider = jsonProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get this beacon node's listening port.",
+      tags = {TAG_NETWORK},
+      description = "Returns the beacon node's listening port.",
```

Comment:
should we expand this to be something like the libp2p port or something? lighthouse says 

> Requests the TCP port that the client's libp2p service is listening on.

i guess I'm mainly asking because it's specifically our p2p port not our http port...

Reply:
added